pytest_fixtures/satellite_auth.py

```diff
@@ -145,3 +146,60 @@ def enroll_idm_and_configure_external_auth():
         f"--server {settings.ipa.hostname_ipa} "
         f"--realm {domain.upper()} -U"
     )
+
+
+@fixture(scope='session')
+def enroll_ad_and_configure_external_auth():
+    """Enroll Satellite Server to an AD Server."""
+    packages = (
+        'sssd adcli realmd ipa-python-compat krb5-workstation '
+        'samba-common-tools gssproxy nfs-utils ipa-client'
+    )
+    realm = settings.ldap.realm
+    workgroup = realm.split(".")[0]
+
+    default_content = f"[global]\nserver = unused\nrealm = {realm}"
+    keytab_content = (
+        f"[global]\nworkgroup = {workgroup}\nrealm = {realm}"
+        f"\nkerberos method = system keytab\nsecurity = ads"
+    )
+
+    # install the required packages
+    run_command(cmd=f'yum -y --disableplugin=foreman-protector install {packages}')
+
+    # update the AD name server
+    run_command(cmd="chattr -i /etc/resolv.conf")
+    result = run_command(
+        cmd="awk -v search='nameserver' '$0~search{print NR; exit}' /etc/resolv.conf"
+    )
+    line_number = int(''.join(result))
+    run_command(
+        cmd=f'sed -i "{line_number}i nameserver {settings.ldap.nameserver}" /etc/resolv.conf'
+    )
+    run_command(cmd="chattr +i /etc/resolv.conf")
+
+    # join the realm
+    run_command(cmd=f"echo {settings.ldap.password} | realm join -v {realm}")
+    run_command(cmd="touch /etc/ipa/default.conf")
+    run_command(cmd=f'echo "{default_content}" > /etc/ipa/default.conf')
+    run_command(cmd=f'echo "{keytab_content}" > /etc/net-keytab.conf')
+
+    # gather the apache id
+    result = run_command(cmd="id apache")
+    _str = ''.join(result).split('(apache)')[0]
+    id_apache = int(re.search(r'\d+', _str).group(0))
```

Comment:
This whole dance can be skipped if you use `id -u`:
```suggestion
    id_apache = run_command(cmd="id -u apache")
```

Reply:
yeah, whole dance was to get uid, thanks for that suggestion (good Linux skill )  :-)